### PR TITLE
complete log.py

### PR DESCRIPTION
**Pull request by moohebat**

### CODE REVIEW

File: transactions/common/log.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import urllib3
+from elasticsearch import Elasticsearch
+import traceback, sys
+import pandas as pd
+
+ES = Elasticsearch(
+        ['https://search-iprice-production-7-pluvqb7oq6wnsisr4k6j62qtlq.ap-southeast-1.es.amazonaws.com/'])
```
Comment (heiwen) [inline transactions/common/log.py]: please use https://es.production.ipricegroup.com:443

File: transactions/common/log.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+def get_data(log_index, id_array):
+    result = []
+    urllib3.disable_warnings()
+    try:
+        for i in range(0, len(id_array), 100):
+            r = ES.search(index=log_index, size=100, body={"query": {"terms": {"_id": id_array[i:i + 100]}}})
+            result += r['hits']['hits']
+    except Exception:
+        print traceback.print_exc()
```
Comment (heiwen) [inline transactions/common/log.py]: As mentioned previously, don't catch "Exception" in general, only specific ones

File: transactions/common/log.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import traceback, sys
+import pandas as pd
+
+ES = Elasticsearch(
+        ['https://search-iprice-production-7-pluvqb7oq6wnsisr4k6j62qtlq.ap-southeast-1.es.amazonaws.com/'])
+
+
+def get_data(log_index, id_array):
+    result = []
+    urllib3.disable_warnings()
```
Comment (heiwen) [inline transactions/common/log.py]: As mentioned previously, do this once in the script, not on every function call

File: transactions/common/log.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            row['_source']['store']['url'] if row['_source'].has_key('store') else '',
+            delete_query_params(row['_source']['exitUrl']) if row['_source'].has_key('exitUrl') else '',
+            delete_query_params(row['_source']['landingUrl']) if row['_source'].has_key('landingUrl') else '',
+            row['_source']['category']['url'].replace("'", ' ') if row['_source'].has_key('category') else '',
+        ]
+        format_data.append(temp)
+    data = pd.DataFrame(format_data,
+                            columns=['ipg:trackingId', 'ipg:sku', 'ipg:created', 'ipg:position', 'ipg:name',
+                                     'ipg:brand', 'ipg:store',
+                                     'ipg:exitURL', 'ipg:landingURL', 'ipg:url'])
```
Comment (heiwen) [inline transactions/common/log.py]: As mentioned previously, no need to store stuff in a temporary array, just put it into the DF directly

File: transactions/common/log.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    return data
+
+
+def delete_query_params(value):
+    if value is not None:
+        return value[0:value.rfind('/') + 1].replace("'", ' ')
+    else:
+        return ''
```
Comment (heiwen) [inline transactions/common/log.py]: As mentioned previously, use a proper urllib to parse and remove the query params

Comment (moohebat) [inline transactions/common/log.py]: Can you discuss this one more.

**Pull request closed without merging**